org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/A4Solution.java

```diff
@@ -1305,6 +1516,49 @@ private static void rename(A4Solution frame, PrimSig s, Map<Sig,List<Tuple>> nex
 
     // ===================================================================================================//
 
+
+    /**
+     * Solve for the solution if not solved already simply using the lowerbound of
+     * each relation as its value. For trace solutions should be called iteratively,
+     * and will build on the information from the previous steps.
+     */
+    // [HASLab]
+    A4Solution solve(final A4Reporter rep, A4Solution pre_sol, int loop) throws Err, IOException {
```

Comment:
this is the method now called, iteratively, by the static reader; it reads a state at a time, and the solution is built incrementally